lib/install/management/finder.go

```diff
@@ -281,22 +280,55 @@ func (d *Dispatcher) searchVCHsPerDC(dc *object.Datacenter) ([]*vm.VirtualMachin
 	d.session.Finder.SetDatacenter(dc)
 
 	var vchs []*vm.VirtualMachine
-	if pools, err = d.session.Finder.ResourcePoolList(d.op, "*"); err != nil {
-		if _, ok := err.(*find.NotFoundError); ok {
-			return vchs, nil
-		}
+
+	pools, err = d.listResourcePools("*")
+	if err != nil {
 		err = errors.Errorf("Failed to search resource pools for datacenter %q: %s", dc.InventoryPath, err)
 		return nil, err
 	}
 
+	vchs = d.searchVCHsFromPools(pools)
+	return vchs, nil
+}
+
+// listResourcePool outputs a list of all resource pools under a compute path
+// retries on ObjectNotFound error because sometimes it's due to concurrent modifications of resource pools
+func (d *Dispatcher) listResourcePools(path string) ([]*object.ResourcePool, error) {
+	var err error
```

Comment:
any reason to define these types separately?
Why wouldn't just use:
```
pools, err := d.session.Finder.ResourcePoolList(d.op, path)
```

Reply:
No actually...I don't remember why I did this...